### ISSUE TITLE
Memory usage regression after Config API merge

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-framework#6670]
Loading the default theme homepage took 10MB peak memory on alpha4, and 27MB peak memory on alpha5. Most likely culprit is the new config system.

![image](https://cloud.githubusercontent.com/assets/111025/23490187/b2ffd016-ff5b-11e6-800c-69a3c4d921ce.png)

```
$ php -v
PHP 5.6.29 (cli) (built: Dec 19 2016 23:48:06) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
    with Xdebug v2.5.0, Copyright (c) 2002-2016, by Derick Rethans
```

```
opcache.blacklist_filename => no value => no value
opcache.consistency_checks => 0 => 0
opcache.dups_fix => Off => Off
opcache.enable => On => On
opcache.enable_cli => Off => Off
opcache.enable_file_override => Off => Off
opcache.error_log => no value => no value
opcache.fast_shutdown => 0 => 0
opcache.file_update_protection => 2 => 2
opcache.force_restart_timeout => 180 => 180
opcache.inherited_hack => On => On
opcache.interned_strings_buffer => 4 => 4
opcache.load_comments => 1 => 1
opcache.log_verbosity_level => 1 => 1
opcache.max_accelerated_files => 2000 => 2000
opcache.max_file_size => 0 => 0
opcache.max_wasted_percentage => 5 => 5
opcache.memory_consumption => 64 => 64
opcache.optimization_level => 0xffffffff => 0xffffffff
opcache.preferred_memory_model => no value => no value
opcache.protect_memory => 0 => 0
opcache.restrict_api => no value => no value
opcache.revalidate_freq => 2 => 2
opcache.revalidate_path => Off => Off
opcache.save_comments => 1 => 1
opcache.use_cwd => On => On
opcache.validate_permission => Off => Off
opcache.validate_root => Off => Off
opcache.validate_timestamps => On => On
```

**Comment by kinglozzer** [comment]
I’ve done a quick bit of digging, I see a tonne of YAML parser/transformer related stuff on every request, instead of just `dev/build` or `?flush`. The `configcache` directory is also empty

**Comment by kinglozzer** [comment]
More digging, the cache never manages to write to the filesystem. It’s got something to do with serialization [here](https://github.com/silverstripe/silverstripe-config/blob/master/src/Collections/MemoryConfigCollection.php#L253-L264). I can’t `var_export` this as it complains about circular references, could that be causing it to fail to serialize?

**Comment by tractorcow** [comment]
The issue is that we don't assign a logger to these caches. @ingo can we update our logger injecter service to assign a default error logger, so these errors are captured somewhere?

**Comment by tractorcow** [comment]
Follow up to hidden caching errors at https://github.com/silverstripe/silverstripe-framework/pull/6693.

Once this is merged we can close this issue.

**Comment by kinglozzer** [comment]
I did notice that the memory usage for a “clean” cache build (i.e. `rm -rf silverstripe-cache`) is higher than alpha-4, so I think there may still be more improvements to be made.

IIRC a significant portion of it appeared to be down to `ExtensionMiddleware:: getExtraConfig()`, but I never got any further in looking into whether anything changed behaviourally there.

**Comment by ingo** [comment]
Just know that I'm not the Ingo you are looking for :)

**Comment by tractorcow** [comment]
I do that a lot sorry, I ment @chillu ;_;

**Comment by tractorcow** [comment]
Closed with #6693
